File: TB2S2016_CSharp/TB2S2016_CSharp/Funcionario.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TB2S2016_CSharp
{
    public class Funcionario : Pessoa, IAutenticavel
    {
        public string SenhaInterna { get; set; }

        public string SenhaExterna { get; set; }

        public override void setNome(string nome, string sobrenome)
        {
            this.nome = nome + " " + sobrenome;
        }

        public double Salario { get; set; }
        public virtual double Bonificacao
        {
            get { return this.Salario * 1.2; }
        }

        public Funcionario(string cpf, string nome)
        {
            this.Cpf  = nome;
            this.Nome = nome;
        }

        bool IAutenticavel.Autentica(string senha)
        {
            return this.SenhaInterna.Equals(senha);
        }
    }
}

File: TB2S2016_Java/src/tb2s2016_java/Funcionario.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package tb2s2016_java;

/**
 *
 * @author antonio.lopes
 */
public class Funcionario extends Pessoa
    implements Autenticavel {
    
    private String senhaInterna;

    public String getSenhaInterna() {
        return senhaInterna;
    }

    public void setSenhaInterna(String senhaInterna) {
        this.senhaInterna = senhaInterna;
    }
            
    private double salario;

    public double getSalario() {
        return salario;
    }

    public void setSalario(double salario) {
        this.salario = salario;
    }
    
    public void setNome(String nome,
            String sobreNome) {
        this.nome = nome + " " + sobreNome;
    }
            
    public double getBonificacao() {
        return this.salario * 1.2;
    }
    
    public Funcionario(String cpf,
            String nome) {
        this.cpf = cpf;
        this.nome = nome;
    }

    @Override
    public boolean autentica(String senha) {
        return this.getSenhaInterna().equals(senha);
    }
}
